torchbiggraph/train_gpu.py

```diff
@@ -495,13 +495,25 @@ def _coordinate_train(self, edges, eval_edge_idxs, epoch_idx) -> Stats:
         edges_lhs = edges.lhs.tensor
         edges_rhs = edges.rhs.tensor
         edges_rel = edges.rel
+        eval_edges_lhs = None
+        eval_edges_rhs = None
+        eval_edges_rel = None
+        full_edges_lhs = None
+        full_edges_rhs = None
+        full_edges_rel = None
         if eval_edge_idxs is not None:
             bucket_logger.debug("Removing eval edges")
             tk.start("remove_eval")
             num_eval_edges = len(eval_edge_idxs)
+            eval_edges_lhs = edges_lhs[eval_edge_idxs]
+            eval_edges_rhs = edges_rhs[eval_edge_idxs]
+            eval_edges_rel = edges_rel[eval_edge_idxs]
             edges_lhs[eval_edge_idxs] = edges_lhs[-num_eval_edges:].clone()
             edges_rhs[eval_edge_idxs] = edges_rhs[-num_eval_edges:].clone()
             edges_rel[eval_edge_idxs] = edges_rel[-num_eval_edges:].clone()
+            full_edges_lhs = edges_lhs
+            full_edges_rhs = edges_rhs
```

Comment:
Mapping into a new edge-set could be expensive memory wise. I'd find a way to avoid